beacon-chain/powchain/log_processing.go

```diff
@@ -346,11 +346,21 @@ func (s *Service) processPastLogs(ctx context.Context) error {
 	}
 
 	s.latestEth1Data.LastRequestedBlock = currentBlockNum
-	currentState, err := s.beaconDB.HeadState(ctx)
+	b, err := s.beaconDB.HeadBlock(ctx)
+	if err != nil {
+		return err
+	}
+	if b == nil || b.Block == nil {
+		return nil
```

Comment:
The decision to return `nil` instead of `error` is to follow the similar pattern as below. 

`PrunePendingDeposits` only gets called if `currentState != nil`